src/Umbraco.Core/Services/ContentTypeServiceBase.cs

```diff
@@ -71,4 +73,887 @@ internal IEnumerable<IContentTypeBase> GetContentTypesForXmlUpdates(params ICont
 
         }
     }
+
+    internal abstract class ContentTypeServiceBase<TItem, TService> : ContentTypeServiceBase
+        where TItem : class, IContentTypeComposition
+        where TService : class, IContentTypeServiceBase<TItem>
+    {
+        protected ContentTypeServiceBase(IDatabaseUnitOfWorkProvider provider, ILogger logger, IEventMessagesFactory eventMessagesFactory)
+            : base(provider, logger, eventMessagesFactory)
+        {
+            _this = this as TService;
+            if (_this == null) throw new Exception("Oops.");
+        }
+
+        private readonly TService _this;
+
+        public static event TypedEventHandler<TService, SaveEventArgs<TItem>> Saving;
```

Comment:
I'm not 100% sure about these static generic events on a base class. I know that these work since we have a similar thing with CacheRefresherBase<T> but this does mean that people can subscribe to events in two ways:

`ContentTypeServiceBase<IContentType, IContentTypeService>.Saving += ...`

or

`ContentTypeService.Saving += ...`

I'm not positive what the ramifications are of this. Static events are ugly/nasty but it's the way we've been working to-date and it's too breaking of a change to shy away from doing this for v8. The other option is to put the events on the actual classes we want them exposed. I know that is more code replication but ....

@zpqrtbnk  thoughts?